### ISSUE TITLE
`filename-case` shouldn't report on directives like "use client"

### ISSUE THREAD

**Issue by yyueniao** [sindresorhus/eslint-plugin-unicorn#2200]
I am working on a NextJs project with App Router, which need to use `use client` [directives](https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs) .

This will be reported by `filename-case` rule.

Rule related:
`filename-case`

```js
"use client" // should pass
import React from "react";
```

**Comment by RealDyllon** [comment]
This error/warning stems from the filename, and reports it at line 1 of the file. From my testing, this isn't affected by the content of the first line. 

Could you share a repo where this can be reproduced??

**Comment by fisker** [comment]
`filename-case` is checking file name. there is no "correct line"...

**Comment by fisker** [comment]
The whole file?
